app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -211,62 +224,72 @@ private function parseReferralCode($referralCode)
-     * If all else fails, create the user .
+     * If userId param is given, check for valid user.
+     * Otherwise check by job record email, mobile.
+     * TODO: Move this to DRY with TurboVote imports (if we keep it).
+     * @see https://www.pivotaltracker.com/n/projects/2019429/stories/164114650
      *
-     * @TODO - If we have a northstar id in the referral code, then we probably don't need to make a call to northstar for the full user object.
-     *
-     * @return array
+     * @param string
+     * @return NorthstarUser
      */
-    private function getOrCreateUser($record, $values)
+    private function getUser($userId)
     {
-        $user = null;
-
-        $userFieldsToLookFor = [
-            'id' => isset($values['northstar_id']) && !empty($values['northstar_id']) ? $values['northstar_id'] : null,
-            'email' => isset($record['Email address']) && !empty($record['Email address']) ? $record['Email address'] : null,
-            'mobile' => isset($record['Phone']) && !empty($record['Phone']) ? $record['Phone'] : null,
-        ];
-
-        foreach ($userFieldsToLookFor as $field => $value)
-        {
-            if ($value) {
-                info('getting user with the '.$field.' field', [$field => $value]);
-                $user = gateway('northstar')->asClient()->getUser($field, $value);
+        if ($userId) {
+            $user = gateway('northstar')->asClient()->getUser('id', $userId);
+            if ($user && $user->id) {
+                return $user;
             }
-
-            if ($user) {
-                break;
+        }
+        if ($this->recordEmail) {
+            $user = gateway('northstar')->asClient()->getUser('email', $this->recordEmail);
+            if ($user && $user->id) {
+                return $user;
             }
         }
+        if (!$this->recordMobile) {
+            return null;
+        }
+        return gateway('northstar')->asClient()->getUser('mobile', $this->recordMobile);
+    }
 
-        if (is_null($user)) {
-            $userData = [
-                'email' => $record['Email address'],
-                'mobile' => $record['Phone'],
-                'first_name' => $record['First name'],
-                'last_name' => $record['Last name'],
-                'addr_street1' => $record['Home address'],
-                'addr_street2' => $record['Home unit'],
-                'addr_city' => $record['Home city'],
-                'addr_state' => $record['Home state'],
-                'addr_zip' => $record['Home zip code'],
-                'source' => env('NORTHSTAR_CLIENT_ID'),
-            ];
+    /**
+     * Creates new user from job record.
+     *
+     * @return NorthstarUser
+     */
+    private function createUser()
+    {
+        $record = $this->record;
+        $userData = [
+            'email' => $this->recordEmail,
+            'mobile' => $this->recordMobile,
```

Comment:
Does this approach work? Looking to avoid hardcoding the `record['Email address']` and `record['Phone']` array keys everywhere with the `$this->recordEmail` and `$this->recordMobile` properties... but then also doesn't seem right to even pass the `$record` parameter into this function at all (as we did before) - which is why I've removed.

Reply:
I wonder if it makes sense to parse the whole CSV record into a sanitized in-memory representation first, so instead of referencing `$this->recordEmail` or `$this->record['Email address']` everywhere, we handle all that messy business up front:

```php
class RockTheVoteRecord {
    public function __construct($record)
    {
        $this->email = $record['Email address'];
        $this->mobile = $record['Phone'];
        $this->first_name = $record['First name'],
        // and so on...
    }
}

class CreateRockTheVotePostInRogue implements ShouldQueue
    public function __construct($record)
    {
        $this->record = new RockTheVoteRecord($record);
    }

    // then anywhere we need these properties, we can use
    // the "standard" names we parsed above…
```

Reply:
I like this idea... but also want to avoid adding even more changes outside of the initial scope of this PR... as that gets even closer to rewriting the entire file. Maybe I'll add a `RockTheVote` class per your example with `record, `email` and `mobile` properties, with a TODO to parse the rest in a future PR?

Reply:
Yeah, that makes good sense to me!